app/uk/gov/hmrc/soletraderidentificationfrontend/models/SoleTraderDetailsMatching.scala

```diff
@@ -36,4 +39,53 @@ object SoleTraderDetailsMatching {
 
   case object DeceasedCitizensDetails extends SoleTraderDetailsMatchFailure
 
+  val KnownFactsMatchingResultKey = "identifiersMatch"
+  val SuccessfulMatchKey = "SuccessfulMatch"
+  val NotEnoughInfoToMatchKey = "NotEnoughInformationToMatch"
+  val DetailsMismatchKey = "DetailsMismatch"
+  val NinoNotDeclaredButFoundKey = "NinoNotDeclaredButFound"
+  val NinoNotFoundKey = "NinoNotFound"
+  val DeceasedCitizensDetailsKey = "DeceasedCitizensDetails"
+
+  implicit val format: Format[SoleTraderDetailsMatchResult] = new Format[SoleTraderDetailsMatchResult] {
+    override def writes(soleTraderMatchingResult: SoleTraderDetailsMatchResult): JsValue = {
+      val knownFactsMatchingResultString = soleTraderMatchingResult match {
+        case SuccessfulMatch => SuccessfulMatchKey
+        case NotEnoughInformationToMatch => NotEnoughInfoToMatchKey
+        case DetailsMismatch => DetailsMismatchKey
+        case NinoNotDeclaredButFound => NinoNotDeclaredButFoundKey
+        case NinoNotFound => NinoNotFoundKey
+        case DeceasedCitizensDetails => DeceasedCitizensDetailsKey
+      }
+
+      JsString(knownFactsMatchingResultString)
+    }
+
+    override def reads(json: JsValue): JsResult[SoleTraderDetailsMatchResult] =
+    //      json.validate[String].collect(JsonValidationError("Invalid Sole Trader Matching Result")) {
```

Comment:
Why are we leaving this commented out?

Reply:
The new reads I wrote caters for Boolean (any old journeys in progress) and the new SoleTraderDetailsMatchResult. Until the change is live and has been for a day there might still be journey's with a boolean stored. After that time we can go back to the simpler reads though so thought I'd leave it commented to it made the next task easier

Reply:
Happy to remove it though and then the other subtask to remove references to the old way of storing it can update the writes